starlette/staticfiles.py

```diff
@@ -161,7 +161,7 @@ def lookup_path(
         self, path: str
     ) -> typing.Tuple[str, typing.Optional[os.stat_result]]:
         for directory in self.all_directories:
-            full_path = os.path.realpath(os.path.join(directory, path))
+            full_path = os.path.abspath(os.path.join(directory, path))
             directory = os.path.realpath(directory)
             if os.path.commonprefix([full_path, directory]) != directory:
                 # Don't allow misbehaving clients to break out of the static files
```

Comment:
As far as I understand `os.path.realpath` will follow symbolic links and here we will stop it from going outside of statics directory.
Instead we can build the path with `abspath` and not follow symlinks (yet) and prevent breaking outside of directory and if file is a symlink it will do `os.stat` and by default it will follow symlinks.

Reply:
Why not change line 165 `realpath` to `abspath`?  If the directory is a symbolic link, line 166 will fail.

Reply:
Can we add a test with the scenario @scientificRat mentioned?

Reply:
sure, I'll take a look.

Reply:
Not sure if I got the point correctly, but I added the tests for this, but I think this change is not needed.